recipes/slow-buffer-to-buffer-alloc-unsafe-slow/src/workflow.ts

```diff
@@ -0,0 +1,350 @@
+import type { Edit, SgRoot, SgNode } from "@codemod.com/jssg-types/main";
+import {
+	getNodeImportStatements,
+	getNodeImportCalls,
+} from "@nodejs/codemod-utils/ast-grep/import-statement";
+import { getNodeRequireCalls } from "@nodejs/codemod-utils/ast-grep/require-call";
+import { resolveBindingPath } from "@nodejs/codemod-utils/ast-grep/resolve-binding-path";
+
+/**
+ * Main entry point that orchestrates all SlowBuffer → Buffer transformations
+ * @param root - The AST root node to transform
+ * @returns The transformed code as a string, or null if no changes were made
+ */
+export default function transform(root: SgRoot): string | null {
+	const rootNode = root.root();
+	const edits: Edit[] = [];
+
+	// Process transformations in order:
+	// 1. CommonJS require statements
+	processRequireStatements(root, edits);
+	// 2. ESM import statements
+	processImportStatements(root, edits);
+	// 3. Dynamic import statements
+	processDynamicImportStatements(root, edits);
+	// 4. Usage patterns (constructor/function calls)
+	processSlowBufferUsage(rootNode, edits);
+
+	if (!edits.length) return null;
+	return rootNode.commitEdits(edits);
+}
+
+/**
+ * Handles CommonJS require('buffer') transformations
+ * Transforms patterns like:
+ * - const SlowBuffer = require('buffer').SlowBuffer → const Buffer = require('buffer').Buffer
+ * - const { SlowBuffer, ... } = require('buffer') → const { Buffer, ... } = require('buffer')
+ * @param root - The AST root to search for require statements
+ * @param edits - Array to collect edit operations
+ */
+function processRequireStatements(root: SgRoot, edits: Edit[]): void {
```

Comment:
This function and the other 2 are almost exactly the same. Could they be DRY'd up?

```ts
type StatementType = 'import-dynamic' | 'import-static' | 'require';

const nodeGetterMap = {
  'import-dynamic': …,
  'import-static': …
  'require': getNodeRequireCalls,
} as const satisfies Record<StatementType, Function>;

function processStatements(
  root: …,
  edits: …,
  type: StatementType,
) {
  const statements = nodeGetterMap[type](root, 'buffer');
  // I think you can keep the rest the same
}
```